src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStoreV1.swift

```diff
@@ -0,0 +1,758 @@
+				return
+			}
+		}
+	}
+
+	private func registerToDidBecomeActiveNotification() {
+		NotificationCenter.default.addObserver(self, selector: #selector(didBecomeActiveNotification), name: UIApplication.didBecomeActiveNotification, object: nil)
+	}
+
+	@objc
+	private func didBecomeActiveNotification(_ notification: Notification) {
+		_ = cleanup()
+	}
+
+	private func fetchContactPersons(for date: String, in database: FMDatabase) -> Result<[DiaryContactPerson], SQLiteErrorCode> {
+		var contactPersons = [DiaryContactPerson]()
+
+		let sql = """
+				SELECT ContactPerson.id AS contactPersonId, ContactPerson.name, ContactPersonEncounter.id AS contactPersonEncounterId
+				FROM ContactPerson
+				LEFT JOIN ContactPersonEncounter
+				ON ContactPersonEncounter.contactPersonId = ContactPerson.id
+				AND ContactPersonEncounter.date = ?
+				ORDER BY ContactPerson.name COLLATE NOCASE ASC, contactPersonId COLLATE NOCASE ASC
+			"""
+
+		do {
+			let result = try database.executeQuery(sql, values: [date])
+
+			while result.next() {
+				let encounterId = result.longLongInt(forColumn: "contactPersonEncounterId")
+				let contactPerson = DiaryContactPerson(
+					id: Int(result.int(forColumn: "contactPersonId")),
+					name: result.string(forColumn: "name") ?? "",
+					encounterId: encounterId == 0 ? nil : Int(encounterId)
+				)
+				contactPersons.append(contactPerson)
+			}
+		} catch {
+			Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+			return .failure(SQLiteErrorCode(rawValue: database.lastErrorCode()) ?? SQLiteErrorCode.unknown)
+		}
+
+		return .success(contactPersons)
+	}
+
+	private func fetchLocations(for date: String, in database: FMDatabase) -> Result<[DiaryLocation], SQLiteErrorCode> {
+		var locations = [DiaryLocation]()
+
+		let sql = """
+				SELECT Location.id AS locationId, Location.name, LocationVisit.id AS locationVisitId
+				FROM Location
+				LEFT JOIN LocationVisit
+				ON Location.id = LocationVisit.locationId
+				AND LocationVisit.date = ?
+				ORDER BY Location.name COLLATE NOCASE ASC, locationId COLLATE NOCASE ASC
+			"""
+
+		do {
+			let result = try database.executeQuery(sql, values: [date])
+
+			while result.next() {
+				let visitId = result.longLongInt(forColumn: "locationVisitId")
+				let contactPerson = DiaryLocation(
+					id: Int(result.int(forColumn: "locationId")),
+					name: result.string(forColumn: "name") ?? "",
+					visitId: visitId == 0 ? nil : Int(visitId)
+				)
+				locations.append(contactPerson)
+			}
+		} catch {
+			Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+			return .failure(SQLiteErrorCode(rawValue: database.lastErrorCode()) ?? SQLiteErrorCode.unknown)
+		}
+
+		return .success(locations)
+	}
+
+	private func updateDiaryDays(with database: FMDatabase) -> DiaryStoringVoidResult {
+		var diaryDays = [DiaryDay]()
+
+		for index in 0...userVisiblePeriodInDays {
```

Comment:
I think this would give us index 0 through 14, so 15 days, but according to the tech spec I think it should be 14 days including today.

```suggestion
		for index in 0..<userVisiblePeriodInDays {
```

Reply:
yes right